moshi/src/main/java/com/squareup/moshi/ClassFactory.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2011 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi
+
+import com.squareup.moshi.internal.Util
+import java.io.ObjectInputStream
+import java.io.ObjectStreamClass
+import java.lang.reflect.InvocationTargetException
+import kotlin.Throws
+
+/**
+ * Magic that creates instances of arbitrary concrete classes. Derived from Gson's UnsafeAllocator
+ * and ConstructorConstructor classes.
+ *
+ * @author Joel Leitch
+ * @author Jesse Wilson
+ */
+internal abstract class ClassFactory<T> {
+  @Throws(InvocationTargetException::class, IllegalAccessException::class, InstantiationException::class)
```

Comment:
Remove this

Reply:
There is java code still requiring the Throws, it can be removed once this is merged https://github.com/square/moshi/pull/1470